dygraph/tsm/train.py

```diff
@@ -168,27 +232,36 @@ def train(args):
         train_config.TRAIN.batch_size = int(train_config.TRAIN.batch_size /
                                             bs_denominator)
 
-        train_reader = KineticsReader(mode="train", cfg=train_config)
+        train_reader = UCF101Reader(name="TSM", mode="train", cfg=train_config)
 
         train_reader = train_reader.create_reader()
         if use_data_parallel:
+            #(data_parallel step4/6)
             train_reader = fluid.contrib.reader.distributed_batch_reader(
                 train_reader)
 
+        # 6. train loop
         for epoch in range(train_config.TRAIN.epoch):
             video_model.train()
             total_loss = 0.0
             total_acc1 = 0.0
             total_acc5 = 0.0
             total_sample = 0
+            t_last = time.time()
+            # 6.1 for each batch, call model() , backward(), and minimize()
             for batch_id, data in enumerate(train_reader()):
+                t1 = time.time()
                 x_data = np.array([item[0] for item in data])
                 y_data = np.array([item[1] for item in data]).reshape([-1, 1])
 
                 imgs = to_variable(x_data)
                 labels = to_variable(y_data)
                 labels.stop_gradient = True
+
+                t2 = time.time()
                 outputs = video_model(imgs)
+                t3 = time.time()
+
                 loss = fluid.layers.cross_entropy(
                     input=outputs, label=labels, ignore_index=-1)
                 avg_loss = fluid.layers.mean(loss)
```

Comment:
copy avg_loss to a new variable , and output(print) it instead of avg_loss, in avoid to print avg_loss after scale_loss function, which is already divided by the number of cards